packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -5453,19 +5467,20 @@ export default class Meeting extends StatelessWebexPlugin {
           this.setNetworkStatus(NETWORK_STATUS.CONNECTED);
           this.reconnectionManager.iceReconnected();
           this.statsAnalyzer.startAnalyzer(this.mediaProperties.webrtcMediaConnection);
+          this.hasMediaConnectionConnectedAtLeastOnce = true;
           break;
         case ConnectionState.Disconnected:
           this.setNetworkStatus(NETWORK_STATUS.DISCONNECTED);
-          this.reconnectionManager.waitForIceReconnect().catch(() => {
+          this.reconnectionManager.waitForIceReconnect().catch(async () => {
             LoggerProxy.logger.info(
               'Meeting:index#setupMediaConnectionListeners --> state DISCONNECTED, automatic reconnection timed out.'
             );
 
-            connectionFailed();
+            await connectionFailed();
```

Comment:
what's the point of adding these awaits? We're inside an event handler and the code that emits the events will not wait for the event handler's returned promise - the event handler's return value is ignored, so having these awaits makes the person reading the code think that something will wait for these, but that's not true, so I think it's better not to have them

Reply:
Removed